test/src/unit-gs.cc

```diff
@@ -0,0 +1,644 @@
+/**
+ * @file   unit-gs.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017-2021 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests for GCS API filesystem functions. (gs extension)
+ */
+
+#ifdef HAVE_GCS
+
+#include "catch.hpp"
+#include "tiledb/common/thread_pool.h"
+#include "tiledb/sm/config/config.h"
+#include "tiledb/sm/filesystem/gcs.h"
+#include "tiledb/sm/global_state/unit_test_config.h"
+#include "tiledb/sm/misc/math.h"
+#include "tiledb/sm/misc/time.h"
+#include "tiledb/sm/misc/utils.h"
+
+#include <sstream>
+
+using namespace tiledb::common;
+using namespace tiledb::sm;
+
+struct GCSFx {
```

Comment:
If you rename this struct so its not duplicated, CI should pass

Reply:
Thank you!